catroid/src/main/java/org/catrobat/catroid/ui/MainMenuActivity.java

```diff
@@ -87,28 +89,54 @@ protected void onCreate(Bundle savedInstanceState) {
 		PreferenceManager.setDefaultValues(this, R.xml.preferences, true);
 		ScreenValueHandler.updateScreenWidthAndHeight(this);
 
-		int oldPrivacyPolicyHash = PreferenceManager.getDefaultSharedPreferences(this)
+		int oldPrivacyPolicy = PreferenceManager.getDefaultSharedPreferences(this)
 						.getInt(AGREED_TO_PRIVACY_POLICY_VERSION, 0);
-		int currentPrivacyPolicyHash = getResources().getString(R.string.dialog_privacy_policy_text)
-						.hashCode();
-		if (oldPrivacyPolicyHash == currentPrivacyPolicyHash) {
+
+		if (oldPrivacyPolicy == Constants.CATROBAT_TERMS_OF_USE_ACCEPTED) {
 			loadContent();
 		} else {
-			setContentView(R.layout.privacy_policy_view);
+			View view = View.inflate(this, R.layout.privacy_policy_view, null);
```

Comment:
Please outsource this to an extra method (e.g. showTermsOfUseDialog()) in order to keep the onCreate() method compact.